silk/tools/otns_manager.py

```diff
@@ -339,48 +408,76 @@ def delete_otns_node(self):
     self.grpc_client.delete_node(self.node_id)
     self.node_on_otns = False
 
-  def update_vis_position(self):
+  def update_otns_vis_position(self):
     """Call gRPC client to update the node's visualization position.
     """
     self.logger.debug(
         "moving node {:d} to OTNS at ({:d},{:d})".format(
             self.node_id, self.vis_x, self.vis_y))
-    self.grpc_client.move_node(self.vis_x, self.vis_y, self.node_id)
+    self.grpc_client.move_node(self.node_id, self.vis_x, self.vis_y)
+
+  def update_vis_position(self, x: int, y: int):
+    if x != self.vis_x or y != self.vis_y:
+      self.vis_x = x
+      self.vis_y = y
+      self.update_otns_vis_position()
 
 
 class WpantundOtnsMonitor(signal.Subscriber):
   """OTNS log monitor for wpantund process logs.
 
   Attributes:
     node (OtnsNode): OTNS node instance that handles UDP messaging.
+    otns_manager (OtnsManager): OTNS manager instance the monitor was
+      created from.
   """
 
   node = None
+  otns_manager = None
 
   def process_log_line(self, line: str):
     """Process a single line of wpantund log.
 
     Args:
         line (str): line of log to process.
     """
+    needs_layout_update = False
     status_match = re.search(RegexType.STATUS.value, line)
     if status_match:
       message = status_match.group(2)
 
-      if self.node is not None:
+      if self.node:
         extaddr_match = re.search(RegexType.EXTADDR_STATUS.value, message)
+        role_match = re.search(RegexType.ROLE_STATUS.value, message)
+        child_added_match = re.search(RegexType.CHILD_ADDED_STATUS.value,
+                                      message)
+        child_removed_match = re.search(RegexType.CHILD_REMOVED_STATUS.value,
+                                        message)
         if extaddr_match:
           self.node.update_extaddr(int(extaddr_match.group(1), 16))
+          needs_layout_update = True
+        elif role_match:
+          self.node.update_role(RoleType(int(role_match.group(1))))
+          needs_layout_update = True
+        elif child_added_match:
+          self.node.add_child(int(child_added_match.group(1), 16))
+          needs_layout_update = True
+        elif child_removed_match:
+          self.node.remove_child(int(child_removed_match.group(1), 16))
+          needs_layout_update = True
```

Comment:
maybe not necessary to update layout when nodes are removed? This can make the layout more stable?

Reply:
Child removed means the topology is changed e.g. a node becoming detached, and it goes to an outer circle. This should update layout imo.